Species agnostic

stschiff opened the pull request:
made some changes to make the format more species-agnostic. Addresses #80

stschiff committed:
made some changes to make the format more species-agnostic

stschiff committed:
Merge branch 'dev' into species_agnostic

stschiff committed:
added species as Janno field

stschiff committed:
added reference genome columns

stschiff committed:
m

nevrome committed:
Merge branch 'dev' into species_agnostic

nevrome committed:
changelog

nevrome commented:
I remember us discussing that the reference genome is actually relevant not just on the sample-level, but also for the entire package. Should we address the decision to keep this information on the sample-level somewhere in the schema? Or is it OK to just explain this in the documentation on the website?

stschiff commented:
Hmm, this is a good question. Genotype data as an aggregate of multiple individuals indeed only makes sense if their positions all refer to the same reference genome. So in a sense we could elevate the new columns `Reference_Genome_Assembly` and `Reference_Genome_Assembly_URL` to the package level. What do you think @TCLamnidis ?

stschiff committed:
moved Species and Reference Genome information to yaml

stschiff commented:
OK, so I've now implemented the somewhat radical change and moved `Species` and `Reference_Genome*` columns over to the `POSEIDON.yml` file, which is the first addition to that file in a while. Take a look. I think it makes sense.

nevrome commented:
Yes - this looks OK for me.

The changelog also needs updating now.

stschiff committed:
updated changelog

stschiff commented:
Done

TCLamnidis commented:
While `Reference_Genome_Assembly{,_URL}` should clearly be in the `POSEIDON.yml` level, as they describe an entire package, `Species` imo should actually be a column in the janno file. 
We already have packages containing multiple species, and that's fine as long as their coordinates are on the same assembly.

stschiff committed:
moved Species back to the Janno

stschiff committed:
updated Changelog

stschiff committed:
Merge branch 'dev' into species_agnostic
